trieste/acquisition/function.py

```diff
@@ -493,3 +494,94 @@ def prepare_acquisition_function(
         eta = tf.reduce_min(tf.boolean_mask(mean, is_feasible), axis=0)
 
         return lambda at: expected_improvement(objective_model, eta, at) * constraint_fn(at)
+
+
+class MonteCarloAcquisition(SingleModelAcquisitionBuilder):
+    """
+    Builder for the expected improvement function where the "best" value is taken to be the minimum
+    of the posterior mean at observed points.
+    """
+
+    def __init__(self, eps_shape: [int]):
+
+        super().__init__()
+        assert len(eps_shape) == 3
```

Comment:
This is still WIP. The batch size B should be defined through the rule, not by hand as here, and the number of latent variables L is given by the models. Only the sample size S may be chosen by the user. 
Also, I am unsure how this should be extended to multiple models.